File: android/src/main/java/expo/modules/semaphore/IdentityModule.kt
package expo.modules.semaphore

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.exception.CodedException
import java.io.File
import uniffi.mopro.*

class IdentityModule : Module() {
  private var identity: Identity? = null

  override fun definition() = ModuleDefinition {
    Name("Identity")

    Function("initIdentity") { privateKey: ByteArray ->
      identity = Identity(privateKey)
    }

    Function("commitment") {
      identity?.commitment()
    }

    Function("privateKey") {
      identity?.privateKey()
    }

    Function("secretScalar") {
      identity?.secretScalar()
    }

    Function("toElement") {
      identity?.toElement()
    }
  }
}

File: ios/IdentityModule.swift
import ExpoModulesCore

public class IdentityModule: Module {
  private var identity: Identity? = nil

  public func definition() -> ModuleDefinition {
    Name("Identity")

    Function("initIdentity") { (privateKey: Data) in
      self.identity = Identity(privateKey: privateKey)
    }

    Function("commitment") {
      return self.identity?.commitment()
    }

    Function("privateKey") {
      return self.identity?.privateKey()
    }

    Function("secretScalar") {
      return self.identity?.secretScalar()
    }

    Function("toElement") {
      return self.identity?.toElement()
    }
  }
}
